lib/src/bits.h

```diff
@@ -7,7 +7,18 @@ static inline uint32_t bitmask_for_index(uint16_t id) {
   return (1u << (31 - id));
 }
 
-#if defined _WIN32 && !defined __GNUC__
+#ifdef __TINYC__
+
+static inline uint32_t count_leading_zeros(uint32_t x) {
+	int count = 0;
+	if (x == 0) return 32;
+	x = x - ((x >> 1) & 0x55555555);
+	x = (x & 0x33333333) + ((x >> 2) & 0x33333333);
+	count = (((x + (x >> 4)) & 0x0f0f0f0f) * 0x01010101) >> 24;
+	return count;
```

Comment:
Is this algorithm based on something that's published? Maybe we should include a comment with a link.